### PR TITLE
feat: use functions/package.json

### PR DESCRIPTION
**Pull request by plmercereau**
- use `functions/package.json` if it exists
- else, use `./package.json` if it exists
- else create and use `functions/package.json`

Notes
- the entire Nhost project still needs to be mounted to `/opt/project` (no change)
- does not tackle the `node_modules` challenge - see the `docker-compose.yaml` in the example
- the `package.json` file in this repository is for development purpose only
- it still supports `pnpm`, `yarn` and `npm` (defaults to `pnpm`), but we are only supporting `yarn` in production

### CODE REVIEW

File: example/docker-compose.yaml [hunk 1]
```diff
@@ -119,7 +119,8 @@ services:
       disable: true
     volumes:
       - .:/opt/project
-      - functions_node_modules:/opt/project/node_modules
+      - project_node_modules:/opt/project/node_modules
+      - functions_node_modules:/opt/project/functions/node_modules
```
Comment (elephant3) [inline example/docker-compose.yaml:123]: oh allright, I think I'll do the same in CLI

**Comment by elitan** [comment]
What is the "does not tackle the node_modules challenge"?

**Review by elitan — APPROVED** [review]

**Comment by plmercereau** [comment]
> What is the "does not tackle the node_modules challenge"?

`node_modules` should not be synced between the host file system and the container FS. It has to be done prior to starting the container. Hence the `package.json` resolution should probably be done in the CLI as well

**Review by elephant3 — APPROVED** [review]
LGTM

File: example/docker-compose.yaml (hunk 1 above)
Comment (elephant3) [reply example/docker-compose.yaml:123]: to prevent clashes with non-functions `node_modules`

**Pull request merged**